### PR TITLE
Add implementation for binary command builtins

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Targets #288

### CODE REVIEW

**Comment by neithernut** [comment]
Binary operations should only operate on integer types (any fail for _anything_ else). For boolean values, the logical commands should be used.

**Comment by matthiasbeyer** [comment]
Fixed, please review again. Will be able to do squashing at 11PM CET or something.

File: src/command/binary.c [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    union ws_value_union* args
+) {
+    if (ws_value_get_type(&args->value) == WS_VALUE_TYPE_NONE ||
+            ws_value_get_type(&args[1].value) != WS_VALUE_TYPE_NONE) {
+        // There is no argument or there are more arguments than one
+        return -EINVAL;
+    }
+
+    return bnot(args);
+}
```
Comment (neithernut) [inline src/command/binary.c:72]: Mh... this function does _exactly_ what `bnot()` does (the checks are unneccessary btw.), so why don't you just integrate `bnot()`'s body into this function and use this one instead from the other ones?

File: src/command/binary.c [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            return -EINVAL;
+        }
+
+        intmax_t buf = ws_value_int_get(&args->int_);
+        buf &= ws_value_int_get(&base->int_);
+        ws_value_int_set(&base->int_, buf);
+    }
+
+    return 0;
+}
```
Comment (neithernut) [inline src/command/binary.c:94]: There are residuals from when you also considered boolean values. (the entire `type` stuff. It's not needed any more) Same holds true for all the other functions.

File: src/command/binary.c (hunk 1 above)
Comment (matthiasbeyer) [inline src/command/binary.c:72]: Why are the checks unneccessary? If that's really the case, I am willing to do this, of course.

Comment (neithernut) [inline src/command/binary.c:72]: Because of the checks you perform in `bnot()`.

Comment (matthiasbeyer) [inline src/command/binary.c:72]: Not exactly: This function also checks if there was really just one argument passed here. The `bnot()` function does not (I did this to be able to use to use the `bnot()` function also for the other `not-...` commands).

**Comment by neithernut** [comment]
You could migrat to the new utility macros now.

**Comment by matthiasbeyer** [comment]
Done.

**Comment by neithernut** [comment]
Looks fine now.

**Comment by matthiasbeyer** [comment]
I'm rebasing now.

**Comment by matthiasbeyer** [comment]
Done.

**Comment by neithernut** [comment]
ACK.

**Pull request merged**